src/stores/FiltersHistoryStore.ts

```diff
@@ -174,9 +170,58 @@ class FiltersHistoryStore {
 		const history = await this.indexedDb.getStoreValues<
 			FiltersHistoryType<EventFilterState | MessageFilterState>
 		>(IndexedDbStores.FILTERS_HISTORY);
-
 		this.filterHistory = history.sort(sortByTimestamp);
+		this.initialized = true;
+	};
+
+	@action
+	private addEventHistoryItem = async (newItem: FiltersHistoryType<EventFilterState>) => {
+		const existedFilter = this.eventsHistory.find(({ filters }) =>
+			isEqual(filters, newItem.filters),
+		);
+		if (existedFilter) {
+			existedFilter.timestamp = newItem.timestamp;
```

Comment:
You can update timestamp since you chose timestamp to be key in indexeddb. this will create new filter